react/components/SKUSelector/components/SelectVariationMode.tsx

```diff
@@ -11,19 +11,35 @@ interface VariationSelectModeProps {
   displayOptions: DisplayOption[]
 }
 
+interface IsDisableParams {
+  disabled: boolean,
+  impossible: boolean,
+  available: boolean
+}
+
 const messages = defineMessages({
   selectPlaceholder: {
     id: 'store/sku-selector.select.placeholder',
     defaultMessage: '',
   },
 })
 
+function isDisabled({ disabled, impossible, available }: IsDisableParams) {
+  if (disabled) {
+    if (impossible) return true
+    if (!impossible && !available) return true
+  }
+
+  return false
+}
+
 function SelectVariationMode(props: VariationSelectModeProps) {
   const { intl, selectedItem, displayOptions } = props
 
   const options = displayOptions.map(op => ({
     label: op.label,
     value: op.label,
+    disabled: isDisabled({disabled: op.disabled, impossible: op.impossible, available: op.available}),
```

Comment:
Just a tip, isn't a request change.
```tsx
     .map({ disabled, impossible, available, label }) => ({
        label,
        value: label,
        disabled: isDisabled({ disabled, impossible, available }),
      })
```

Reply:
Can you lint this file?

Reply:
Ah yes, sorry about that, I ran eslint --fix on that file and found 2 warnings and 3 errors.  Should be fine now.